### ISSUE TITLE
decode_solution() error

### ISSUE THREAD

**Issue by ymatsumoto** [recruit-communications/pyqubo#25]
**Describe the bug**
If there are binary product of three or more variables in an expression, decode_solution() outputs an error due to 'KeyError'.

**To Reproduce**
```python
>>> from pyqubo import Sum, Matrix, Vector, Constraint, Placeholder, utils
>>> import functools
>>> import operator
>>> n = 10
>>> x = Matrix('x', n_row=1, n_col=n)
>>> prod = functools.partial(functools.reduce, operator.mul)
>>> exp = prod((1.0 - x[0,j]) for j in range(0, n))
>>> model = exp.compile()
>>> q, offset = model.to_qubo()
>>> s = utils.solve_qubo(q)
>>> decoded_solution, broken, energy = model.decode_solution(s, vartype='BINARY')
KeyError: 'x[0][8]*x[0][3]'
```

**Expected behavior**
Key should be generated as 'x[0][3]*x[0][8]'.

**Comment by kotarotanahashi** [comment]
@ymatsumoto Thank you for reporting the issue! This bug is caused by inconsistent label names. It is solved in #26.

**Comment by ymatsumoto** [comment]
@kotarotanahashi Thank you for your quick code fix! I also confirmed the problem was solved.
